diff -r -d -u a/longlong.h b/longlong.h
--- a/longlong.h	2023-07-29 13:42:16.000000000 +0000
+++ b/longlong.h	2026-03-06 13:28:12.654497273 +0000
@@ -1156,13 +1156,14 @@
 #endif /* i960mx */
 #endif /* i960 */
 
-
 #if defined (__loongarch64) && W_TYPE_SIZE == 64
+#include <stdint.h>
+
 #define umul_ppmm(w1, w0, u, v) \
   do {									\
     UDItype __u = (u), __v = (v);					\
     (w0) = __u * __v;							\
-    (w1) = (unsigned __int128__) __u * __v >> 64;			\
+    (w1) = (__uint128_t) __u * __v >> 64;			\
   } while (0)
 #endif
 
